spring-kafka/src/main/java/org/springframework/kafka/support/LogSupport.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.support;
+
+import org.apache.commons.logging.Log;
+
+/**
+ * Logging utilities.
+ *
+ * @author Gary Russell
+ * @since 2.1.2
+ *
+ */
+public final class LogSupport {
+
+	private LogSupport() {
+		super();
+	}
+
+	/**
+	 * Logging levels.
+	 */
+	public enum Level {
+
+		/**
+		 * Fatal.
+		 */
+		FATAL,
+
+		/**
+		 * Error.
+		 */
+		ERROR,
+
+		/**
+		 * Warn.
+		 */
+		WARN,
+
+		/**
+		 * Info.
+		 */
+		INFO,
+
+		/**
+		 * Debug.
+		 */
+		DEBUG,
+
+		/**
+		 * Trace.
+		 */
+		TRACE
+
+	}
+
+	public static boolean isEnabled(Level level, Log logger) {
+		switch (level) {
+			case FATAL:
+				return logger.isFatalEnabled();
+			case ERROR:
+				return logger.isErrorEnabled();
+			case WARN:
+				return logger.isWarnEnabled();
+			case INFO:
+				return logger.isInfoEnabled();
+			case DEBUG:
+				return logger.isDebugEnabled();
+			case TRACE:
+				return logger.isTraceEnabled();
+		}
+		return false;
+	}
+
+	public static void logAt(Level level, Log logger, String message) {
+		switch (level) {
+			case FATAL:
+				logger.fatal(message);
+				break;
+			case ERROR:
+				logger.error(message);
+				break;
+			case WARN:
+				logger.warn(message);
+				break;
+			case INFO:
+				logger.info(message);
+				break;
+			case DEBUG:
+				logger.debug(message);
+				break;
+			case TRACE:
+				logger.trace(message);
+				break;
+		}
+	}
+
+	public static void logAt(Level level, Log logger, String message, Throwable t) {
```

Comment:
Can't we move these two method to the `LogSupport.Level` to avoid extra argument for `static` methods and have a context and encapsulation with the polymorphism as they are stated in Java. 😄  

Plus pay attention to the logging method of `Log` contract. They are `Object`-based for the `message` argument. I would stay with that.

Thanks

Reply:
Good points, both.